mpeg2/src/temi.rs

```diff
@@ -0,0 +1,275 @@
+use crate::{DecodeError, EncodeError};
+use byteorder::{BigEndian, ByteOrder};
+use core2::io::Write;
+
+const AF_DESCR_TAG_TIMELINE: u8 = 0x04;
+
+#[derive(Debug, Default, PartialEq)]
+pub struct TEMITimelineDescriptor {
+    pub has_timestamp: Option<u8>,
+    pub timescale: u32,
+    pub media_timestamp: u64,
+
+    pub ntp_timestamp: Option<u64>,
+    pub ptp_timestamp: Option<u128>,
+
+    pub has_timecode: Option<u8>,
+    pub drop: bool,
+    pub frames_per_tc_seconds: u16,
+    pub duration: u16,
+    pub time_code: u64,
+
+    pub force_reload: Option<bool>,
+    pub paused: Option<bool>,
+    pub discontinuity: Option<bool>,
+    pub timeline_id: u8,
+}
+
+impl TEMITimelineDescriptor {
+    pub fn encode_len(&self) -> usize {
+        let mut len = 5;
+        if let Some(ts) = self.has_timestamp {
+            // timescale + media_timestamp
+            if ts == 1 {
+                len += 8;
+            } else if ts == 2 {
+                len += 12;
+            }
+        }
+        if self.ntp_timestamp.is_some() {
+            len += 8;
+        } else if self.ptp_timestamp.is_some() {
+            len += 10;
+        }
+        if let Some(timecode) = self.has_timecode {
+            len += 4;
+            if timecode == 1 {
+                len += 3;
+            } else {
+                len += 8;
+            }
+        }
+        len
+    }
+
+    pub fn encode<W: Write>(&self, mut w: W) -> Result<usize, EncodeError> {
+        let mut buf = vec![0u8; self.encode_len()];
+        buf[0] = AF_DESCR_TAG_TIMELINE;
+        if let Some(timestamp) = self.has_timestamp {
+            buf[2] |= timestamp << 6;
+        }
+
+        if self.force_reload.unwrap_or(false) {
+            buf[2] |= 0b0000_0010;
+        }
+        if self.paused.unwrap_or(false) {
+            buf[2] |= 0b0000_0001;
+        }
+        if self.discontinuity.unwrap_or(false) {
+            buf[3] = 0b1000_0000;
+        }
```

Comment:
Shouldn't this be...

```suggestion
        if self.force_reload.unwrap_or(false) {
            buf[2] |= 0b0010_0000;
        }
        if self.paused.unwrap_or(false) {
            buf[2] |= 0b0100_0000;
        }
        if self.discontinuity.unwrap_or(false) {
            buf[3] = 0b1000_0000;
        }
```

?

Reply:
![image](https://user-images.githubusercontent.com/13324159/167704999-b59bed36-bf7c-4dbc-9738-c853fa55587c.png)
I think `paused` is the lowest-order bit in `buf[2]`.

Reply:
Ah, you're right. I read that wrong. 👍